iridium-core-server/src/main/java/software/iridium/api/service/AuthorizationService.java

```diff
@@ -445,4 +441,50 @@ public String proxyAuthorizationRequestToProvider(
     // todo: throw exception for provider not found
     return null;
   }
+
+  @Transactional(propagation = Propagation.REQUIRED)
+  public AccessTokenResponse refreshToken(Map<String, String> params) {
+
+    checkArgument(
+        attributeValidator.isNotBlank(
+            params.getOrDefault(AuthorizationCodeFlowConstants.GRANT_TYPE.getValue(), "")),
+        "grant_type must not be blank");
+
+    checkArgument(
+        attributeValidator.isNotBlank(
+            params.getOrDefault(AuthorizationCodeFlowConstants.CLIENT_ID.getValue(), "")),
+        "Client ID must not be blank");
+
+    checkArgument(
+        attributeValidator.isNotBlank(
+            params.getOrDefault(AuthorizationCodeFlowConstants.REFRESH_TOKEN.getValue(), "")),
+        "Refresh_token must not be blank");
+
+    checkArgument(
+        attributeValidator.equals(
+            params.get(AuthorizationCodeFlowConstants.GRANT_TYPE.getValue()),
+            AuthorizationCodeFlowConstants.REFRESH_TOKEN_GRANT_TYPE.getValue()),
+        "Grant_type value must be set to \"refresh_token\"");
```

Comment:
These `checkArgument`'s should be in a validator class.  You can see examples of them [here](https://github.com/IridiumIdentity/iridium/tree/main/iridium-core-server/src/main/java/software/iridium/api/validator)

Reply:
done

Reply:
These `checkArgument` functions are still in the `AuthorizationService` class.  They should be in a validator class.  I listed examples of them above.